cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/DefaultMetadataStore.java

```diff
@@ -564,15 +578,12 @@ private void execute(TransactionExecutor.Procedure<MetadataDataset> func, Metada
     }, scope);
   }
 
-  private MetadataDataset newMetadataDataset(DatasetFramework dsFramework, MetadataScope scope) {
-    try {
-      return DatasetsUtil.getOrCreateDataset(
-        dsFramework, getMetadataDatasetInstance(scope), MetadataDataset.class.getName(),
-        DatasetProperties.builder().add(MetadataDatasetDefinition.SCOPE_KEY, scope.name()).build(),
-        DatasetDefinition.NO_ARGUMENTS);
-    } catch (Exception e) {
-      throw Throwables.propagate(e);
-    }
+  private MetadataDataset newMetadataDataset(DatasetContext datasetContext,
```

Comment:
there is already a getMetadataDataset() with almost the same signature. Should only have one of them

Reply:
fixed